### ISSUE TITLE
Add key expiration (ttl) 

### ISSUE THREAD

**Issue by ashcrow** [ashcrow/consuloretcd#6]
A request came in asking to support ttl for keys. Both consultant etcd support ttl but in different ways. A single interface should be created so that switching is seamless.

**Comment by ashcrow** [comment]
- **etcd**: https://coreos.com/etcd/docs/2.0.8/api.html
- **consul**: https://github.com/hashicorp/consul/pull/524

**Comment by ashcrow** [comment]
Work started in https://github.com/ashcrow/consuloretcd/tree/ttl

Looked quickly at the consul implementation but it wasn't nearly as clear how to make a session expire a key.

**Comment by ashcrow** [comment]
Took a few times through the doc to realize that the session has to be made first and then the key must be created **with** the session. This seems to work. I'll try to implement it shortly.

### curl example

The following expires the key after 30 seconds.

``` bash
$ curl -H "Content-Type: application/json" -d '{"TTL": "30s", "Behavior": "delete"}' -X PUT http://127.0.0.1:8500/v1/session/create
{ "ID":"ad91aa2d-c562-182f-9938-7a913a32022e"}
$ curl -X PUT -d "test" http://127.0.0.1:8500/v1/kv/test2?acquire=ad91aa2d-c562-182f-9938-7a913a32022e
```

**Comment by ashcrow** [comment]
https://github.com/ashcrow/consuloretcd/commit/689cf0c84ce36c102d3e93d8c6287db39b45f66a implementes sessions for TTL in consul. Needs some clean up work but functionality should be there. Waiting on some user testing.

**Comment by ashcrow** [comment]
User testing came back positive.

**Comment by ashcrow** [comment]
Merged in #7
